### PR TITLE
#1334 Typescript conversion - Avatar, Background, Badge

### PR DESCRIPTION
**Pull request by emil-botha**
- Convert the Avatar, Background and Badge components from Javascript to TypeScript
- Use functional instead of class components
- Implement initial unit tests

### CODE REVIEW

File: packages/bappo-components/src/components/Avatar/types.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+export type AvatarProps = {
+  backgroundColor: string;
```
Comment (Jonathanliu525) [inline packages/bappo-components/src/components/Avatar/types.ts:2]: if backgroundColor is must define, do we still need default value for that

File: packages/bappo-components/src/components/Background/Background.tsx [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-const Background = ({ children }: Props) => {
-  return <StyledView selectable>{children}</StyledView>;
+  children?: string;
+  testID?: string;
 };
 
-Background.defaultProps = {};
-
-export default Background;
+export default function Background({ children, testID }: Props = {}) {
```
Comment (Jonathanliu525) [inline packages/bappo-components/src/components/Background/Background.tsx]: { children, testID }: Props = {}   Hi Emil, should we keep the {} as default value? I am not sure about that, could went me through that when you free? Thanks

File: packages/bappo-components/src/components/Avatar/types.ts (hunk 1 above)
Comment (emil-botha) [reply packages/bappo-components/src/components/Avatar/types.ts:2]: I've removed the default value for the required backgroundColor

File: packages/bappo-components/src/components/Background/Background.tsx (hunk 2 above)
Comment (emil-botha) [reply packages/bappo-components/src/components/Background/Background.tsx]: Thanks, I've also removed the `{}` as default value.

File: packages/bappo-components/src/components/Badge/Badge.tsx [hunk 3]
```diff
@@ -1,26 +1,21 @@
...
 import View from '../../primitives/View';
 
 type Props = {
-  number?: number,
+  number?: number;
+  testID?: string;
 };
 
-const Badge = ({ number }: Props) => {
+export default function Badge({ number, testID }: Props = {}) {
```
Comment (Jonathanliu525) [inline packages/bappo-components/src/components/Badge/Badge.tsx]: Hi Emil, I am sorry I forget to mention  in Badge, there is also a empy object

Comment (emil-botha) [reply packages/bappo-components/src/components/Badge/Badge.tsx]: Thanks Jon, I've removed the empty default value from Badge too.

**Review by Jonathanliu525 — APPROVED** [review]

**Pull request merged**